src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -512,6 +774,31 @@ public FormValidation doCheckScanCredentialsId(@AncestorInPath SCMSourceOwner co
             }
         }
 
+        @Restricted(NoExternalUse.class)
+        public FormValidation doCheckBuildOriginBranch/* arbitrary which one we pick for web method name */(
+            @QueryParameter boolean buildOriginBranch,
+            @QueryParameter boolean buildOriginBranchWithPR,
+            @QueryParameter boolean buildOriginPRMerged,
+            @QueryParameter boolean buildOriginPRUnmerged,
+            @QueryParameter boolean buildForkPRMerged,
+            @QueryParameter boolean buildForkPRUnmerged
+        ) {
+            if (!buildOriginBranch && !buildOriginBranchWithPR && !buildOriginPRMerged && !buildOriginPRUnmerged && !buildForkPRMerged && !buildForkPRUnmerged) {
+                return FormValidation.warning("You need to build something!");
+            }
+            if (buildOriginBranchWithPR && buildOriginPRUnmerged) {
+                return FormValidation.warning("Redundant to build an origin PR both as a branch and as an unmerged PR");
+            }
+            if (buildOriginBranch && !buildOriginBranchWithPR && !buildOriginPRMerged && !buildOriginPRUnmerged && !buildForkPRMerged && !buildForkPRUnmerged) {
+                // TODO in principle we could make doRetrieve populate originBranchesWithPR without actually including any PRs, but it would be more work and probably never wanted anyway.
+                return FormValidation.warning("If you are not building any PRs, all origin branches will be built.");
+            }
+            if (buildOriginPRMerged && buildOriginPRUnmerged || buildForkPRMerged && buildForkPRUnmerged) {
+                return FormValidation.ok("Merged vs. unmerged PRs will be distinguished in the job name.");
```

Comment:
Well, these messages apply to _combinations_ of checkboxes, not a single one.

Perhaps better to rename the method `doCheckBuildForkPRUnmerged` so it appears at the bottom rather than at the top. Will have to play with it to see.